In [1]:
#initalConfig integer < 2^m: listCs the feedback polynomial as a list in same order as in the book.
def lsfr(initialConfig, listCs):
    newConfig = initialConfig
    count = 0
    while True:
        outputBit = newConfig & 1
        xored = 0
        for i in listCs[0:len(listCs)-1]:
            xored = xored ^ ((newConfig >> i) & 1)
        #print (newConfig)
        print (format(newConfig,'0{}b'.format(listCs[-1])), outputBit)
        newConfig = (newConfig >> 1)|(xored << (listCs[-1]-1))
        count = count + 1 
        if (newConfig == initialConfig):
            break
    print('length of Seq:', count)

In [2]:
lsfr(15, [0,1,4])

1111 1
0111 1
0011 1
0001 1
1000 0
0100 0
0010 0
1001 1
1100 0
0110 0
1011 1
0101 1
1010 0
1101 1
1110 0
length of Seq: 15


x^4 + x^1 + 1  : Primitive polynomial generating max-length sequence 15. 

In [3]:
for i in range(1,16):
    lsfr(i, [0,2,4])
    print('\n')
    

0001 1
1000 0
0100 0
1010 0
0101 1
0010 0
length of Seq: 6


0010 0
0001 1
1000 0
0100 0
1010 0
0101 1
length of Seq: 6


0011 1
1001 1
1100 0
1110 0
1111 1
0111 1
length of Seq: 6


0100 0
1010 0
0101 1
0010 0
0001 1
1000 0
length of Seq: 6


0101 1
0010 0
0001 1
1000 0
0100 0
1010 0
length of Seq: 6


0110 0
1011 1
1101 1
length of Seq: 3


0111 1
0011 1
1001 1
1100 0
1110 0
1111 1
length of Seq: 6


1000 0
0100 0
1010 0
0101 1
0010 0
0001 1
length of Seq: 6


1001 1
1100 0
1110 0
1111 1
0111 1
0011 1
length of Seq: 6


1010 0
0101 1
0010 0
0001 1
1000 0
0100 0
length of Seq: 6


1011 1
1101 1
0110 0
length of Seq: 3


1100 0
1110 0
1111 1
0111 1
0011 1
1001 1
length of Seq: 6


1101 1
0110 0
1011 1
length of Seq: 3


1110 0
1111 1
0111 1
0011 1
1001 1
1100 0
length of Seq: 6


1111 1
0111 1
0011 1
1001 1
1100 0
1110 0
length of Seq: 6




x^4 + x^2 + 1: reducible polynomial, length of sequence depends on input 

(x^2 + x + 1)(x^2 -x + 1) Two different length sequences given above for initialization of 1111 and 0110.

In [4]:
for i in range(1,16):
    lsfr(i , [0,1,2,3,4])
    print ('\n')

0001 1
1000 0
1100 0
0110 0
0011 1
length of Seq: 5


0010 0
1001 1
0100 0
1010 0
0101 1
length of Seq: 5


0011 1
0001 1
1000 0
1100 0
0110 0
length of Seq: 5


0100 0
1010 0
0101 1
0010 0
1001 1
length of Seq: 5


0101 1
0010 0
1001 1
0100 0
1010 0
length of Seq: 5


0110 0
0011 1
0001 1
1000 0
1100 0
length of Seq: 5


0111 1
1011 1
1101 1
1110 0
1111 1
length of Seq: 5


1000 0
1100 0
0110 0
0011 1
0001 1
length of Seq: 5


1001 1
0100 0
1010 0
0101 1
0010 0
length of Seq: 5


1010 0
0101 1
0010 0
1001 1
0100 0
length of Seq: 5


1011 1
1101 1
1110 0
1111 1
0111 1
length of Seq: 5


1100 0
0110 0
0011 1
0001 1
1000 0
length of Seq: 5


1101 1
1110 0
1111 1
0111 1
1011 1
length of Seq: 5


1110 0
1111 1
0111 1
1011 1
1101 1
length of Seq: 5


1111 1
0111 1
1011 1
1101 1
1110 0
length of Seq: 5




x^4 + x^3 + x^2 + x + 1 : irreducible but not primitive. Length of sequence: 5 

In [5]:
lsfr(255, [0,1,3,4,8])

11111111 1
01111111 1
00111111 1
00011111 1
00001111 1
10000111 1
01000011 1
00100001 1
10010000 0
11001000 0
11100100 0
01110010 0
00111001 1
10011100 0
01001110 0
00100111 1
00010011 1
10001001 1
01000100 0
00100010 0
10010001 1
01001000 0
10100100 0
01010010 0
00101001 1
00010100 0
10001010 0
01000101 1
10100010 0
11010001 1
01101000 0
10110100 0
11011010 0
11101101 1
01110110 0
00111011 1
00011101 1
10001110 0
01000111 1
00100011 1
00010001 1
00001000 0
10000100 0
01000010 0
10100001 1
11010000 0
11101000 0
11110100 0
11111010 0
11111101 1
11111110 0
length of Seq: 51


2.7: 1+x+x^3+x^4+x^8, Initialization Vector: FF 

first two output bytes: 0000100111001000  || x09C8

Turns out this one is not max-length polynomial.